projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -158,6 +159,13 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
     @Input()
     public data: any[];
 
+    /**
+     * An @Input property that sets the resource strings.
+     * By default it uses EN resources.
+    */
+    @Input()
+    public resourceStrings: IGridResourceStrings = GridResourceStringsEN;
```

Comment:
In the input's setter we must assign the newly passed object with the default English resources. This will ensure that the result object will have all resources even when the user haven't provided all strings.